### ISSUE TITLE
FCN failing to train for ADNI data

### ISSUE THREAD

**Issue by usadiqgriffin** [vkola-lab/brain2020#7]
I am working on a problem using a subset of ADNI data (330 training images) and using your model to train, but I am unable to get good performance on the data:

1: FCN training and validation accuracies always hover around ~50%. The validation confusion matrix indicates that the FCN either classifies all as 0 or 1, and keeps alternating between the two during training.
2: Tried changing learning rate, file_num etc to no use
3: mlp_A which uses FCN and DPMs also produces similar performance (around ~50% over 300 training epochs)
4: mlp_C which uses FCN and clinical features also hovers around that performance
5: I tried to give an easy classification problem to FCN(0s and 1s with linear boundary) and it quickly goes to 100% accuracy which tells me that learning is taking place

Any suggestions you could help me with? Is FCN supposed to do better standalone or does it need more layers? Why do CNN and FCN have 1 as the input number when we are feeding them patches of 47x47?

**Comment by shangranq** [comment]
Hi, FCN is independent of later stage MLP, so when you are debugging the system, I recommend you firstly get reasonable results on FCN model, then move on to MLP part. Regarding you didn't manage to get good performance, (a) is your task the same as ours? (b) if so, what changes you made in the script? 

To answer your last question: (1) our paper suggested this generic FCN approach for disease visualization; you can play with different number of layers to see the response. (2) please elaborate more on "having 1 as input number"
